packages/mui-base/src/TextareaAutosize/TextareaAutosize.tsx

```diff
@@ -122,71 +118,26 @@ const TextareaAutosize = React.forwardRef(function TextareaAutosize(
 
     // Take the box sizing into account for applying this value as a style.
     const outerHeightStyle = outerHeight + (boxSizing === 'border-box' ? padding + border : 0);
-    const overflow = Math.abs(outerHeight - innerHeight) <= 1;
+    const overflowing = Math.abs(outerHeight - innerHeight) <= 1;
 
-    return { outerHeightStyle, overflow };
+    return { outerHeightStyle, overflowing };
   }, [maxRows, minRows, props.placeholder]);
 
-  const updateState = (prevState: State, newState: State) => {
-    const { outerHeightStyle, overflow } = newState;
-    // Need a large enough difference to update the height.
-    // This prevents infinite rendering loop.
-    if (
-      renders.current < 20 &&
-      ((outerHeightStyle > 0 &&
-        Math.abs((prevState.outerHeightStyle || 0) - outerHeightStyle) > 1) ||
-        prevState.overflow !== overflow)
-    ) {
-      renders.current += 1;
-      return {
-        overflow,
-        outerHeightStyle,
-      };
-    }
-    if (process.env.NODE_ENV !== 'production') {
-      if (renders.current === 20) {
-        console.error(
-          [
-            'MUI: Too many re-renders. The layout is unstable.',
-            'TextareaAutosize limits the number of renders to prevent an infinite loop.',
-          ].join('\n'),
-        );
-      }
-    }
-    return prevState;
-  };
-
   const syncHeight = React.useCallback(() => {
-    const newState = getUpdatedState();
+    const textareaStyles = calculateTextareaStyles();
 
-    if (isEmpty(newState)) {
+    if (isEmpty(textareaStyles)) {
       return;
     }
 
-    setState((prevState) => updateState(prevState, newState));
-  }, [getUpdatedState]);
+    const input = inputRef.current!;
+    input.style.setProperty('height', `${textareaStyles.outerHeightStyle}px`);
+    input.style.setProperty('overflow', textareaStyles.overflowing ? 'hidden' : null);
```

Comment:
Why not a straight assignment?

```suggestion
    input.style.height = ${textareaStyles.outerHeightStyle}px`);
    input.style.overflow = textareaStyles.overflowing ? 'hidden' : null;
```

Reply:
> Why not a straight assignment?

Because initially, I prioritized `height` as `important`. Now, we can do straight assignment. 👍